### DISCUSSION TITLE
itwin-fronted3.7.5 =>itwin4.0.3

### DISCUSSION THREAD

**Discussion by VWlf** [iTwin/itwinjs-core#6671] [Q&A]
During the upgrade process from itwin 3.7.5 to itwin 4.0.3, it was discovered during project compilation that optional chains were added in 4.0.3 The relevant code of the error message is as follows:

**Comment by aruniverse** [comment]
Please share the error

**Reply by VWlf** [reply]
Just released, could you please help me take a look

**Comment by VWlf** [comment]
![报错截图](https://github.com/iTwin/itwinjs-core/assets/49948146/08ce814b-4d99-453d-82ec-193ef9caad9f)
I am trying to add a new feature for “?.”or"??" In <babel.config.js> Adaptation, but no effect. By the way, my current project is built by Vue2,Node is 20.12.2
![babel set](https://github.com/iTwin/itwinjs-core/assets/49948146/5938dd83-9a8d-46db-bcea-c498d7d9ad95)
Babel's related dependencies are all above 7.0.0. Please help me identify the problem. If you need more problem descriptions, please contact me. Thank you very much
![babel_version](https://github.com/iTwin/itwinjs-core/assets/49948146/db3d26fd-f9b4-4686-99aa-8a9241644ed4)

**Reply by aruniverse** [reply]
We don't use babel directly, we use a [fork of create-react-app](https://github.com/imodeljs/create-react-app/blob/imodeljs/packages/react-scripts/package.json), you can look at the dependencies we use, and our [config](https://github.com/imodeljs/create-react-app/blob/imodeljs/packages/react-scripts/config/webpack.config.js)

I see you added the plugin for optional chaining, but looks like it was for the propsal, maybe you should be using https://babeljs.io/docs/babel-plugin-syntax-optional-chaining.html

**Reply by VWlf** [reply]
If Babel uses Propsal in version 7.0 and syntax in version 6.0, I have tried both but failed. I will try what you said[https://github.com/imodeljs/create-react-app/blob/imodeljs/packages/react-scripts/package.json](url)
How was this added to the Vue project for use? Was it added in package.json

**Reply by VWlf** [reply]
Sorry, I saw it. It's the link at the back. I'll give it a try. Thank you very much！！

**Comment by VWlf** [comment]
I know the answer, if you're configuring it in a Vue2 project, you'll need to add it to your vue.config.js:
`rules: [
        {
          test: /\.js$/,
          include: /node_modules/,
          use: {
            loader: 'babel-loader'  
          }
        }
      ]`
Install dependencies:
`npm install -g babel-loader @babel/core @babel/preset-env
`
